metal_header/device.h

```diff
@@ -40,6 +44,99 @@ class Device {
 
     void emit_include(std::string d);
 
+    std::string platform_define(node n, std::string suffix);
+    std::string platform_define_offset(node n, std::string suffix);
+
+    Inline *create_inline_dec(std::string name, std::string ret_type, std::string arg) const
```

Comment:
Is there a reason these aren't defined in a .c++ file?

Reply:
> Is there a reason these aren't defined in a .c++ file?

not really, was starting with it and forgot to move into c++. Will see if I can move. Otherwise will do in subsequence merge.